stdlib/scanf.mli

```diff
@@ -92,6 +92,8 @@ type in_channel
    A Scanf.Scanning.in_channel value is also called a {i formatted input
    channel} or equivalently a {i scanning buffer}.
    The type {!Scanning.scanbuf} below is an alias for [Scanning.in_channel].
+   Note that a [Scanning.in_channel] is not thread-safe: concurrent use may
+   produce arbitrary values or exceptions.
```

Comment:
Since this comment is inside the `Scanning` submodule, it will not be directly displayed in the documentation (see [ocaml.org](https://ocaml.org/api/Scanf.html) or [v3.ocaml.org](https://v3.ocaml.org/p/ocaml-base-compiler/4.13.1/doc/Stdlib/Scanf/index.html#module-Scanning) for instance). We will probably still need a global comments on the `Scanf` module but I would propose to do that in a separate PR.